projects/Mallard/src/hooks/use-image-size.tsx

```diff
@@ -0,0 +1,22 @@
+import React, { useState, useEffect, createContext, useContext } from 'react'
+import { ImageSize } from '../../../Apps/common/src'
+import { imageForScreenSize } from 'src/helpers/screen'
+
+const Context = createContext<ImageSize>('phone')
+
+type Props = { children: React.ReactNode }
+export const ImageSizeProvider = ({ children }: Props) => {
+    const [imageSize, setImageSize] = useState<ImageSize | null>(null)
+    useEffect(() => {
+        imageForScreenSize().then(imgSize => {
+            setImageSize(imgSize)
+        })
+    })
+
+    if (imageSize == null) return null
```

Comment:
Should we be returning null here? I think we should always have the presence of the context provider, otherwise the consumer has nothing to subscribe to which might lead to some inconsistencies?

Reply:
Which is what it does, because the consumers are in the `children`. So the image size isn't loaded yet we don't render the rest. Ideally we should parallelise everything, which would be achieved by having a single store for the app global state and using `Promise.all` to resolve all the initialisation data, but that's something we can work towards iteratively.